packages/cli/src/transformers.ts

```diff
@@ -861,21 +923,115 @@ export function transformSourceMap(sourceMap: string, extension: string) {
   return JSON.stringify(updatedSourceMapObject);
 }
 
+/**
+ * Transform the source map URL to match the new file extension of the source
+ * file.
+ *
+ * @param content - The content of the source file.
+ * @param extension - The new file extension.
+ * @param declarationExtension - The new file extension for declaration files.
+ * @returns The transformed content.
+ */
+function transformSourceMapUrl(
+  content: string,
+  extension: string,
+  declarationExtension: string,
+) {
+  // This is a bit hacky, but TypeScript doesn't provide a way to transform
+  // the source map comment in the source file.
+  return content
+    .replace(
+      /^\/\/# sourceMappingURL=(.*)\.js\.map$/mu,
+      `//# sourceMappingURL=$1${extension}.map`,
+    )
+    .replace(
+      /^\/\/# sourceMappingURL=(.*)\.d\.ts\.map$/mu,
+      `//# sourceMappingURL=$1${declarationExtension}.map`,
+    );
+}
+
+/**
+ * The regular expression to match dynamic imports. The aim of `ts-bridge` is to
+ * avoid regular expressions as much as possible, but this is a special case
+ * where we need to match dynamic imports created by the TypeScript compiler,
+ * after AST transformation.
+ *
+ * This regular expression matches dynamic imports, but only if they are not
+ * part of a larger identifier. This is to avoid matching `import.meta.url`,
+ * `foo.import`, etc.
+ *
+ * The following imports will be matched:
+ *
+ * ```ts
+ * import('./foo.js')
+ * import('./foo/bar.js').SomeClass
+ * ```
+ *
+ * The following imports will not be matched:
+ *
+ * ```ts
+ * import.meta.url
+ * foo.import('./foo.js')
+ * _import('./foo.js')
+ * ```
+ */
+const DYNAMIC_IMPORT_REGEX = /(?<![\w.])import\(['"](\..+)['"]\)/gu;
+
+/**
+ * Transform the dynamic imports in the declaration file to use the new file
+ * extension.
+ *
+ * @param content - The content of the declaration file.
+ * @param extension - The new file extension.
+ * @param parentUrl - The URL of the parent module.
+ * @param system - The TypeScript system.
+ * @param verbose - Whether to show verbose output.
+ * @returns The transformed content.
+ */
+export function transformDeclarationImports(
+  content: string,
+  extension: string,
+  parentUrl: string,
+  system: System,
+  verbose: boolean,
+) {
+  return content.replace(
+    DYNAMIC_IMPORT_REGEX,
```

Comment:
I don't like that we need to do this at all, but I couldn't find a better solution here. TypeScript doesn't seem to call the transformer functions with the actual AST that's converted to code and written to disk.